internal/kubelet/client.go

```diff
@@ -167,7 +169,7 @@ func (p *saClientProvider) BuildClient() (Client, error) {
 	tr := defaultTransport()
 	tr.TLSClientConfig = &tls.Config{
 		RootCAs:            rootCAs,
-		InsecureSkipVerify: true,
+		InsecureSkipVerify: p.insecureSkipVerify,
```

Comment:
It seems that since its introduction the serviceAccount auth method has been setting this value to `True`.  To keep this as a non-breaking change, to we need to ensure that this value is true by default for serivceAccount auth?

Reply:
I think setting `insecure_skip_verify` to `true` by default only for the service account auth type will make the configuration confusing for users with very little advantage to them. This change in behavior is likely to not impact most k8s users. Production grade clusters usually follow the k8s recommended practice of bootstrapping the Kubelet server with TLS, serving certificate signed by the cluster’s root CA. Fwiw, I have checked that Kubelet serves properly signed certs (with node ip address set in the SAN field in addition to the node name) in eks, gke, kubeadm, and kops. We can keep this PR/issue open for a little longer to see if any users have any concerns. And I can update the changelog to mark this breaking instead of bug_fix to make it more visible to the users. Let me know if you’d still like to explore making the default different for SA auth.

Reply:
Although technically breaking, your response makes me now feel ok about it. Since this won't affect most users we can call it a bug fix.

Reply:
I changed it to breaking since I found some discrepancy in the SAN of kubelet certs in older vs newer gke clusters. The older version seems to use the IP and FQDN of the instance in the kubelet cert instead of the k8s node name.